File: 백준/Gold/1016. 제곱 ㄴㄴ 수/제곱 ㄴㄴ 수.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        long min = Long.parseLong(st.nextToken());
        long max = Long.parseLong(st.nextToken());

        boolean[] isSquare = new boolean[(int)(max - min + 1)];

        for (long i = 2; i * i <= max; i++) {
            long square = i * i;

            long start = min / square;
            if (min % square != 0) start++;

            for (long j = start; square * j <= max; j++) {
                isSquare[(int)(square * j - min)] = true;
            }
        }

        int count = 0;
        for (boolean b : isSquare) {
            if (!b) count++;
        }

        System.out.println(count);
    }
}

File: 백준/Gold/1016. 제곱 ㄴㄴ 수/제곱 ㄴㄴ 수.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.StringTokenizer

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val st = StringTokenizer(br.readLine())
    val min = st.nextToken().toLong()
    val max = st.nextToken().toLong()

    val isSquare = BooleanArray((max - min + 1).toInt())

    var i = 2L
    while (i * i <= max) {
        val square = i * i

        var start = min / square
        if (min % square != 0L) start++

        var j = start
        while (square * j <= max) {
            isSquare[(square * j - min).toInt()] = true
            j++
        }

        i++
    }

    println(isSquare.count { !it })
}